pkg/apis/keycloak/v1alpha1/keycloakbackup_types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KeycloakBackupSpec defines the desired state of KeycloakBackup
+// +k8s:openapi-gen=true
+type KeycloakBackupSpec struct {
+	Restore bool            `json:"restoreFrom,omitempty"`
+	AWS     KeycloakAWSSpec `json:"aws,omitempty"`
+}
+
+// KeycloakAWSSpec defines the desired state of KeycloakBackupSpec
+// +k8s:openapi-gen=true
+type KeycloakAWSSpec struct {
+	EncryptionKeySecretName  string `json:"encryptionKeySecretName"`
+	AwsCredentialsSecretName string `json:"awsCredentialsSecretName"`
+	Schedule                 string `json:"schedule"`
+}
+
+type BackupStatusPhase string
+
+var (
+	BackupPhaseNone        BackupStatusPhase
+	BackupPhaseReconciling BackupStatusPhase = "reconciling"
+	BackupPhaseCreated     BackupStatusPhase = "created"
+	BackupPhaseRestored    BackupStatusPhase = "restored"
+	BackupPhaseFailing     BackupStatusPhase = "failing"
+)
+
+// KeycloakBackupStatus defines the observed state of KeycloakBackup
+// +k8s:openapi-gen=true
+type KeycloakBackupStatus struct {
```

Comment:
Would be good to add in the secondary resources here also, the same as the Keycloak CR

Reply:
Fixed.